src/OrchardCore.Modules/Orchard.Liquid/Drops/QueriesDrop.cs

```diff
@@ -32,6 +32,11 @@ public override object BeforeMethod(string method)
                 return ((IEnumerable<JObject>)result).Select(x => new JTokenDrop(x)).ToArray();
             }
 
+            if (result is IEnumerable<object>)
```

Comment:
Oh, that was one of my suggestions in your original PR. If you prefer i can do this.

But when `ReturnDocuments = true`, are the returned objects always `JObject`? Those from the `IEnumerable<object>` returned by the query.